src/js/components/channels/twilio-channel-content.jsx

```diff
@@ -1,34 +1,134 @@
 import React, { Component } from 'react';
+import { updateTwilioAttributes, resetTwilioAttributes } from '../../services/integrations-service';
 
 import { ReactTelephoneInput } from '../../lib/react-telephone-input';
 
-export class TwilioChannelContent extends Component {
-    state = {
-        numberValid: false
+import { connect } from 'react-redux';
+import isMobile from 'ismobilejs';
+
+export class TwilioChannelContentComponent extends Component {
+
+    linkTwilioNumber = () => {
+        const {state: {formattedNumber}} = this.refs.telInput;
+        updateTwilioAttributes({
+            number: formattedNumber,
+            linkState: 'pending'
+        });
     }
 
+    onRetry = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
+
+    onChangeNumber = () => {
+        updateTwilioAttributes({
+            linkState: 'unlinked',
+            number: '',
+            numberValid: false
+        });
+    }
+
+    onStartTexting = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
+
+    onSendText = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
 
     onNumberValid = () => {
-        this.setState({
+        updateTwilioAttributes({
             numberValid: true
         });
     }
 
     onNumberInvalid = () => {
-        this.setState({
+        updateTwilioAttributes({
             numberValid: false
         });
     }
 
+    componentWillUnmount() {
+        resetTwilioAttributes();
+    }
+
     render() {
-        return <div>
-                   <ReactTelephoneInput ref='telInput'
-                                        defaultCountry='ca'
-                                        onChange={ this.handleInputChange }
-                                        onValid={ this.onNumberValid }
-                                        onInvalid={ this.onNumberInvalid }
-                                        preferredCountries={ ['ca', 'us'] }
-                                        onBlur={ this.handleInputBlur } />
-               </div>;
+        let iconStyle = {};
+        if (this.props.settings.linkColor) {
```

Comment:
I see a lot of `this.props.something` all over this method, can we get them from the start with `const { ...} = this.props` and avoid repetition?